pkg/cloudinstances/cloud_instance_group.go

```diff
@@ -51,45 +51,22 @@ func (c *CloudInstanceGroup) NewCloudInstance(instanceId string, newGroupName st
 	}
 	node := nodeMap[instanceId]
 
-	if newGroupName == currentGroupName {
+	if status == CloudInstanceStatusReady {
 		c.Ready = append(c.Ready, cm)
-		cm.Status = CloudInstanceStatusReady
 	} else {
 		c.NeedUpdate = append(c.NeedUpdate, cm)
-		cm.Status = CloudInstanceStatusNeedsUpdate
 	}
 
 	if node != nil {
 		cm.Node = node
+		cm.Status = status
 	} else {
 		klog.V(8).Infof("unable to find node for instance: %s", instanceId)
 		cm.Status = CloudInstanceStatusNotJoined
 	}
 	return nil
 }
 
-// NewDetachedCloudInstance creates a new CloudInstance for a detached instance
-func (c *CloudInstanceGroup) NewDetachedCloudInstance(instanceId string, nodeMap map[string]*v1.Node) error {
```

Comment:
This change is incorrect. It means a detached instance that hasn't joined won't be recognized as detached.